command/volume_status_csi.go

```diff
@@ -33,8 +33,26 @@ func (c *VolumeStatusCommand) csiStatus(client *api.Client, id string) int {
 		c.Ui.Error(fmt.Sprintf("No volumes(s) with prefix or ID %q found", id))
 		return 1
 	}
+
+	var ns string
+
+	if len(vols) == 1 {
+		id = vols[0].ID
+		ns = vols[0].Namespace
+	}
+
+	// List sorts by CreateIndex, not by ID, so we need to search for
+	// exact matches but account for multiple exact ID matches across
+	// namespaces
 	if len(vols) > 1 {
-		if (id != vols[0].ID) || (c.allNamespaces() && vols[0].ID == vols[1].ID) {
+		exactMatchesCount := 0
```

Comment:
How do you feel about declaring this outside the if? If you did that it would decrease the number of braced indents and make the logic flow a little a little clearer.

Reply:
I'm not sure I see how it would decrease indents? It's both declared and used entirely inside the `if`

Reply:
Oh wait, I see what you mean... remove the length check entirely and iterate unconditionally, so that we _always_ check for exact matches. That works.

Reply:
I've added a comment to the `len(vols) == 1` case to make that clear.

Reply:
Ahh nm. This was a function GH formatting things. Once I opened the file I see how that's not possible. Sorry for that.